### PR TITLE
fix: set the status of image

### PR DESCRIPTION
**Pull request by DOO-DEV**

### CODE REVIEW

**Comment by arshamalh** [comment]
@DOO-DEV @ezzoddin 
The purpose of `getImageStatus` was not only to see image status, but also to see each image is in use by which containers (which can be more than one), that's the reason for complicated code.
Although the implemented code previously was not exactly what I expected, but the algorithm in general was OK.

File: docker/images.go [hunk 1]
```diff
@@ -39,26 +39,23 @@ func (d *docker) ImageRemove(ctx context.Context, imageID string, force, pruneCh
 }
 
 func (d *docker) getImageStatus(ctx context.Context, image types.ImageSummary) (status string) {
-	if len(image.RepoTags) == 0 {
+	if image.RepoTags[0] == "<none>:<none>" {
```
Comment (arshamalh) [inline docker/images.go]: There are cases that there might be no RepoTags, in that case, if you want to access the first element in the slice, we'll face a panic.
image

File: docker/images.go [hunk 2]
```diff
@@ -39,26 +39,23 @@ func (d *docker) ImageRemove(ctx context.Context, imageID string, force, pruneCh
...
-		if cont.ImageID != image.ID {
-			status = string(entities.ImageStatusUnUsed)
-		} else {
-			newSlice := newImgs[image.ID]
-			if newSlice == nil {
-				newSlice = make([]string, 0)
-			}
-			newSlice = append(newSlice, cont.ID)
-			newImgs[image.ID] = newSlice
+	if image.Containers == 0 {
```
Comment (arshamalh) [inline docker/images.go]: According to the docs:
> Number of containers .... **This size is not calculated by default, and depends on which API endpoint is used. `-1` indicates that the value has not been set / calculated.**

So this number will always be "-1" cause we didn't set it.

But if you're still sure that I might be wrong, let's discuss it in a meeting or something.

**Comment by arshamalh** [comment]
#8 related issue.

File: docker/images.go (hunk 2 above)
Comment (DOO-DEV) [reply docker/images.go]:
```go
type ImageSummary struct {

	// Number of containers using this image. Includes both stopped and running
	// containers.
	//
	// This size is not calculated by default, and depends on which API endpoint
	// is used. `-1` indicates that the value has not been set / calculated.
	//
	// Required: true
	Containers [int64](https://pkg.go.dev/builtin#int64) `json:"Containers"`
```

yes you  are rigth. `-1` means no api still set/calc the number of containers at this moment

File: docker/images.go (hunk 1 above)
Comment (DOO-DEV) [reply docker/images.go]: I cant simulate a state that the `image.RepoTag` be zero length. can you show me the image that has no tag in your machin?
and the state `dangling` is when we have `none:noe` tag? am i right?

Comment (DOO-DEV) [reply docker/images.go]: There is a case that all of our image has tag and we don't have any container. then the logic return status with empty string.
what you think about this code:
```go
func (d *docker) getImageStatus(ctx context.Context, image types.ImageSummary) (status string) {
	if len(image.RepoTags) == 0 {
		status = string(entities.ImageStatusUnUsedDangling)
		return
	}

	if image.RepoTags[0] == "<none>:<none>" {
		status = string(entities.ImageStatusUnUsedDangling)
		return
	}
	containers, _ := d.cli.ContainerList(ctx, types.ContainerListOptions{})
	if len(containers) == 0 {
		status = string(entities.ImageStatusUnUsed)
		return
	}
	newImgs := make(map[string][]string)
	for _, cont := range containers {
		if cont.ImageID != image.ID {
			status = string(entities.ImageStatusUnUsed)
		} else {
			newSlice := newImgs[image.ID]
			if newSlice == nil {
				newSlice = make([]string, 0)
			}
			newSlice = append(newSlice, cont.ID)
			newImgs[image.ID] = newSlice

			status = string(entities.ImageStatusInUse)
		}
	}
```

**Pull request merged**